nesta/packages/dq/general.py

```diff
@@ -0,0 +1,132 @@
+import pandas as pd
+import numpy as np
+import itertools
+
+def missing_values(input):
+    """missing_values
+
+    Args:
+        input (:obj:`iter` of :obj:`obj`): A sequence of objects.
+
+    Returns:
+        missing_count_df (:obj:`pandas.core.series.Series`): A series with column names (index) and missing value frequency (column)
+
+
+    """
+    df = pd.DataFrame(input)
+    missing_count_df = df.isnull().sum().sort_values(ascending=False)
+    return missing_count_df
+
+def missing_value_percentage_column_count(input):
+    """missing_value_percentage_column_count
+
+    Args:
+        input (:obj:`dict`): A dictionary of
+
+    Returns:
+        out_counts (:obj:`pandas.core.series.Series`): A series with mising value percentage ranges (index) and column frequency (column)
+
+    """
+
+    missing_bins = [i for i in range(100+1) if i%10 == 0]
+    out= pd.cut((pd.Series(input)/total_length).apply(lambda x: round(x*100,5)),
+                                                         bins=missing_bins, include_lowest=True)
+    out_counts = out.value_counts(sort=False)
+
+    return out_counts
+def missing_value_column_count(input):
+    """missing_value_column_count
+
+    Args:
+        input (:obj:`iter` of :obj:`obj`): A sequence of objects.
+
+    Returns:
+        out (:obj:`tuple`): A tuple of two lists- first list with the frequency, second list the bins
+
+    """
+
+    df = pd.DataFrame(input)
+
+    data_frame_missing = df.isnull().sum() / df.shape[0]
+    out = np.histogram(data_frame_missing, bins=range(0, 101, 10))
+    return out
+
+def missing_value_count_pair_both(input):

```

Comment:
This works great. I can't remember what we discussed, but it might be best to return the result as a dataframe or perhaps to have the option to choose between outputting as a dataframe or json

Reply:
I didn't look down the page! Looks like you could just have an argument and that can trigger `melt_dict_to_df`